pinot-segment-local/src/main/java/org/apache/pinot/segment/local/upsert/PartialUpsertHandler.java

```diff
@@ -65,15 +73,18 @@ public PartialUpsertHandler(Schema schema, Map<String, UpsertConfig.Strategy> pa
    * @return a new row after merge
    */
   public GenericRow merge(GenericRow previousRecord, GenericRow newRecord) {
-    for (Map.Entry<String, PartialUpsertMerger> entry : _column2Mergers.entrySet()) {
-      String column = entry.getKey();
-      if (!previousRecord.isNullValue(column)) {
-        if (newRecord.isNullValue(column)) {
-          newRecord.putValue(column, previousRecord.getValue(column));
-          newRecord.removeNullValueField(column);
-        } else {
-          newRecord.putValue(column,
-              entry.getValue().merge(previousRecord.getValue(column), newRecord.getValue(column)));
+    for (String column : previousRecord.getFieldToValueMap().keySet()) {
```

Comment:
shouldn't the iteration be done on the columns in the newRecord?

Reply:
the new column will always be present in `newRecord` (which is equivalent to an overwrite strategy). if there is an update to the same primary key, then the previous record will have that new column value and iterate through this loop. So, it should be fine to iterate through the columns of the previous record, right?

Reply:
Makes sense.